core/polyaxon/connections/gcp/gcs.py

```diff
@@ -339,29 +339,16 @@ def delete(self, key, bucket_name=None, workers=0):
         if not bucket_name:
             bucket_name, key = self.parse_gcs_url(key)
 
-        results = self.list(bucket_name=bucket_name, key=key, delimiter="/")
-        if not any([results["prefixes"], results["blobs"]]):
-            self.delete_file(key=key, bucket_name=bucket_name)
-
-        # Delete directories
-        for prefix in sorted(results["prefixes"]):
-            prefix = os.path.join(key, prefix)
-            # Download files under
-            self.delete(key=prefix, bucket_name=bucket_name)
-
+        blobs = self.connection.list_blobs(bucket_name, prefix=key)
```

Comment:
`bucket.list_blobs` is deprecated and directly using `self.connection.list_blobs` with a bucket name also remove the need to do a GET request in `get_bucket()`.